internal/extproc/translator/openai_openai.go

```diff
@@ -151,5 +179,26 @@ func (o *openAIToOpenAITranslatorV1ChatCompletion) extractUsageFromBufferEvent()
 			o.buffered = nil
 			return
 		}
+		if o.backendName != "" && o.modelName != "" {
+			now := time.Now()
+			if !o.firstTokenSent {
+				o.firstTokenSent = true
+				o.metrics.FirstTokenLatency.WithLabelValues(o.backendName, o.modelName).
+					Observe(now.Sub(o.requestStart).Seconds())
+			} else {
+				// Calculate the time between tokens.
+				// Since we are only interested in the time between tokens, and openai streaming is by chunk,
+				// we can calculate the time between tokens by the time between the last token and the current token, divided by the number of tokens.
+				// And in some cases, the number of tokens can be 0, so we need to check for that.
+				div := tokenUsage.OutputTokens
+				if div == 0 {
+					div = 1
+				}
+				itl := now.Sub(o.lastTokenTime).Seconds() / float64(div)
```

Comment:
Can extract to an util function as it is used by multiple translators

Reply:
sure, will do